packages/devtools_app/lib/src/screens/debugger/breakpoint_manager.dart

```diff
@@ -175,6 +185,86 @@ class BreakpointManager with DisposerMixin {
     ]);
   }
 
+  Future<List<Breakpoint>> _getBreakpointsForIsolate(String isolateId) async {
+    final isolate = await _service.getIsolate(isolateId);
+    if (isolate.id != _isolateRefId) {
+      // Current request is obsolete.
+      return [];
+    }
+
+    // Ignore attempts from DWDS to re-establish breakpoints because DevTools is
+    // now in charge of re-establishing breakpoints:
+    final connectedToDwds =
+        serviceConnection.serviceManager.connectedApp?.isDartWebAppNow ?? false;
+    if (connectedToDwds) return [];
+
+    return isolate.breakpoints ?? [];
+  }
+
+  Future<void> _updateBreakpoints({
+    required List<Breakpoint> breakpoints,
+    required String isolateId,
+  }) async {
+    _breakpoints.value = breakpoints;
+    // Build _breakpointsWithLocation from _breakpoints.
+    await Future.wait(
+      _breakpoints.value.map(breakpointManager.createBreakpointWithLocation),
+    ).then((list) {
+      if (isolateId != _isolateRefId) {
+        // Current request is obsolete.
+        return;
+      }
+      _breakpointsWithLocation.value = list.toList()..sort();
+    });
+  }
+
+  Future<void> _setUpBreakpoints({
+    required List<BreakpointAndSourcePosition> breakpoints,
+    required IsolateRef isolateRef,
+  }) async {
+    final scriptUriToRef = await _scriptRefsForBreakpoints(
+      breakpoints: breakpoints,
+      isolateRef: isolateRef,
+    );
+
+    for (final breakpoint in breakpoints) {
+      final newScriptRef = scriptUriToRef[breakpoint.scriptUri];
+      final breakpointLine = breakpoint.line;
+
+      if (newScriptRef?.id != null && breakpointLine != null) {
```

Comment:
nit: pull `newScriptRef?.id` into a a constant so that it will be type promoted as non nullable below.

Reply:
Done!